Inclusion #2: Plotting of Logistic Regression of PCA'd data w/ MRSA and CA datasets

MrBones1102 opened the pull request:
This is the second split of the Regression PR. It includes plotting for logistic regression of the MRSA and CA datasets.

aarmey commented on mrsa_ca_rna/figures/figure03_RegressionScore.py:
```diff
@@ -0,0 +1,102 @@
+    ca_df = whole_data[whole_data.obs["disease"] == "Candidemia"].to_df()
+
+    datasets = {"MRSA": mrsa_df, "MRSA+CA+Healthy": combined_df, "CA": ca_df}
+    performance_dict = datasets.copy()
```
Why do you need a copy of this? Can't you just add information to `datasets`?

aarmey commented on mrsa_ca_rna/figures/figure03_RegressionScore.py:
```diff
@@ -0,0 +1,102 @@
+    datasets = {"MRSA": mrsa_df, "MRSA+CA+Healthy": combined_df, "CA": ca_df}
+    performance_dict = datasets.copy()
+
+    comp_arr = np.arange(1, components + 1)
```
Just make this where you need it.

aarmey commented on mrsa_ca_rna/figures/figure03_RegressionScore.py:
```diff
@@ -0,0 +1,102 @@
+            performance.append(nested_performance)
+
+        matrix = np.array([comp_arr.astype(int), performance]).T
+        performance_df = pd.DataFrame(
```
Rather than concatenate, then create a dataframe, you can create a dataframe from a dictionary. Something like `pd.Dataframe({"Comp": np.arange, "Performance": performance})`. This is much more readable.

aarmey commented on mrsa_ca_rna/figures/figure03_RegressionScore.py:
```diff
@@ -0,0 +1,102 @@
+
+            # slice our X_data to our current components and set y_data to be MRSA status from whole data
+            X_data = scores_df.iloc[:, : i + 1]
+            y_data = whole_data.obs.loc[whole_data.obs["disease"] == "MRSA", "status"]
```
Because y_data is always the same, I'd move it out of the loop.

aarmey commented on mrsa_ca_rna/figures/figure04.py:
```diff
@@ -0,0 +1,72 @@
+    weighted_rna = df.copy()
+    # for every patient, multiply their gene expression values by the weights
+    for pat in range(weighted_rna.shape[0]):
+        weighted_rna.iloc[pat, :] = weighted_rna.iloc[pat, :].values * weights
```
I think you're going to have a much easier time inspecting the weights of the model, than weighting the underlying data and looking for patterns...

aarmey commented on mrsa_ca_rna/figures/figure04.py:
```diff
@@ -0,0 +1,72 @@
+    for pat in range(weighted_rna.shape[0]):
+        weighted_rna.iloc[pat, :] = weighted_rna.iloc[pat, :].values * weights
+    totals = []
+    for col in weighted_rna.columns:
```
This looks like something that can be vectorized. Can't you do a `np.sum(X, axis=0`?

aarmey commented on mrsa_ca_rna/figures/figure04.py:
```diff
@@ -0,0 +1,72 @@
+    totals = []
+    for col in weighted_rna.columns:
+        totals.append(weighted_rna[col].sum())
+    total_df = pd.DataFrame(
```
Yeah... I think you're trying to extract information from the dataset that should be evident in the model parameters themselves.

MrBones1102 commented:
@aarmey I incorporated your suggested changes to figure03, it makes a lot more sense now. I removed figure04 entirely because I feel like it just wasn't showing something useful. Incidentally, I had started a new branch for making figures related to dissecting the components more directly using the weights, which I feel like was figure04's intended purpose, so I would likely have replaced figure04 with these new figures anyway.
Plus, figure04 was very heavy computationally.

Side note: I apologize if this method of interaction is improper. Please let me know if you have a more preferred method or format for receiving rationale or communication.

aarmey commented:
@MrBones1102 communication on here is always great, so long as you keep your pull requests reasonably organized! It's much easier to discuss code changes because comments and changes can be organized next to each other.

aarmey reviewed (approved):
Ready to merge.
